### ISSUE TITLE
[EA] Null pointer exception may occur here. False positive.

### ISSUE THREAD

**Issue by Chemaclass** [kalessil/phpinspectionsea#1814]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended), 4.0.7.1 |
| Language level | PHP 7.4  |

Current behaviour
--
When using "Person.ageNullable" property, then there is a complain about a possible null although the method `Person->getName()` has a non-nullable return type of `/** @return Age */`.

<img width="858" alt="Screenshot 2022-02-22 at 18 03 10" src="https://user-images.githubusercontent.com/5256287/155181619-51677587-aa06-435d-ae99-31c994fdff25.png">

The reason seems to be that it's getting the return type from the property `/** @var Age|null */` and not from the method `/** @return Age */`.

<img width="924" alt="Screenshot 2022-02-22 at 18 05 34" src="https://user-images.githubusercontent.com/5256287/155182008-807f03e1-eba5-43cb-991f-b053f93d81fa.png">

Expected behaviour
--

I would expect to don't have this warning of "Possible NULL" when the method has a specific non-nullable return type.

<img width="727" alt="Screenshot 2022-02-22 at 18 06 47" src="https://user-images.githubusercontent.com/5256287/155182212-8b0621ef-c2d1-42cb-894b-233fea9faf54.png">

Environment details
---
This applies to all PHP versions. But especially from PHP >=7.4

The code used to reproduce the example
```php
<?php declare(strict_types = 1);

class Age
{
    private int $value;

    public static function fromInt(int $age): self
    {
        if ($age < 0) {
            throw new \RuntimeException('Impossible');
        }
        $self = new self();
        $self->value = $age;
        return $self;
    }

    public function value(): int
    {
        return $this->value;
    }
}

class Person
{
    /** @var Age|null */
    private $ageNullable;

    /** @var Age */
    private $ageForced;

    public static function withAge(int $age): self
    {
        $self = new self();
        $self->age = Age::fromInt($age);
        return $self;
    }

    /** @return Age */
    public function getAge()
    {
        return $this->ageNullable;
    }
}

class MainExample
{
    public static function main(): void
    {
        $person = Person::withAge(28);
        $age = $person->getAge();

        self::printAge($age); # <---- THIS IS A FALSE POSITIVE
    }

    private static function printAge(Age $age): void
    {
        echo 'Age value: ' . $age->value();
    }
}
```

**Comment by mike-zukowski** [comment]
Exactly the same issue with older Symfony code:

```
abstract class ContainerAwareCommand extends Command implements ContainerAwareInterface
{
    /**
     * @var ContainerInterface|null
     */
    private $container;

    /**
     * @return ContainerInterface
     *
     * @throws \LogicException
     */
    protected function getContainer()
```

then in any extending class:

```
$this->getContainer()->getParameter('kernel.environment')
```

ends with `Null pointer exception may occur here`.

**Comment by alsii** [comment]
Same thing occurs in Doctrine:

```php
        $criteria = Criteria::create()
            ->where(Criteria::expr()->eq('active', 1)) // <---- THIS IS A FALSE POSITIVE
        ;
```
![Снимок экрана 2022-04-04 180501](https://user-images.githubusercontent.com/5807252/161585972-cd116dbb-23f5-422b-9cf5-d414988ca844.png)

Doctrine Code:

```php
    /**
     * Returns the expression builder.
     *
     * @return \Doctrine\Common\Collections\ExpressionBuilder
     */
    public static function expr()
    {
        if (self::$expressionBuilder === null) {
            self::$expressionBuilder = new ExpressionBuilder();
        }

        return self::$expressionBuilder;
    }
```

**Comment by ea-inspections-team** [comment]
Thank you for reporting, @Chemaclass! This inspection appears to be somewhat unreliable, so we are disabling it in e1263f8b0607a1f50598cc49c6dbfdcdd73f4baa for now.
